src/fides/api/api/v1/endpoints/admin.py

```diff
@@ -27,23 +28,28 @@ class DBActions(str, Enum):
     ],
     status_code=status.HTTP_200_OK,
 )
-async def db_action(action: DBActions) -> Dict:
+async def db_action(action: DBActions, revision: Optional[str] = "head") -> Dict:
     """
     Initiate one of the enumerated DBActions.
     """
-    action_text = "upgrade"
 
-    if action == DBActions.reset:
-        if not CONFIG.dev_mode:
-            raise HTTPException(
-                status_code=status.HTTP_501_NOT_IMPLEMENTED,
-                detail="Resetting the application database outside of dev_mode is not supported.",
-            )
+    if action == DBActions.downgrade:
+        downgrade_db(CONFIG.database.sync_database_uri, revision=revision)  # type: ignore[arg-type]
+        action_text = "downgrade"
+    else:
+        action_text = "upgrade"
 
-        reset_db(CONFIG.database.sync_database_uri)
-        action_text = "reset"
+        if action == DBActions.reset:
+            if not CONFIG.dev_mode:
+                raise HTTPException(
+                    status_code=status.HTTP_501_NOT_IMPLEMENTED,
+                    detail="Resetting the application database outside of dev_mode is not supported.",
+                )
 
-    await configure_db(CONFIG.database.sync_database_uri)
+            reset_db(CONFIG.database.sync_database_uri)
+            action_text = "reset"
+
+        await configure_db(CONFIG.database.sync_database_uri, revision=revision)
```

Comment:
If existing upgrade fails here, it will fail in the logs but the API will return that it performed successfully

Reply:
nice, new try/catch looking good:

Error: Internal Server Error
--



```json
{
  "detail": "Database downgrade failed: PostgreSQL ENUM type requires a name.. Check server logs for more details"
}
```

Reply:
Not addressed, but also not part of this ticket, so we can defer

Reply:
thank you for the diligence! https://ethyca.atlassian.net/browse/PROD-2118